### ISSUE TITLE
complete metadata

### ISSUE THREAD

**Issue by sebgoa** [serverless/serverless-kubeless#33]
I looked at the other plugins in detail to find the lowest common denominator and cross-checked `kubeless` functionality.

There are a few features that we need to add in `kubeless` and I created:
https://github.com/kubeless/kubeless/issues/215 to track them.

As of today, we need to aim for complete metadata to become:

```
provider:
  name: kubeless
  namespace: default
  runtime: nodejs6.10
  memorySize: 512

functions:
  hello:
    handler: handler.hello
    description: "this will be free from annotation for a function"
    environment:
      variable2: value2
    labels:
      foo: bar
    events:
      - http:
          path: users/create
          method: get
      - trigger: <topic_name>
      - schedule: '*/8 * * * * *'
```

* `namespace` can be added right away https://github.com/serverless/serverless-kubeless/issues/26
* `memorySize` needs support added in kubeless
* `description` needs support added
* `environment` needs support added
* `labels` needs support added
* `events` schedule needs support added but trigger and http already exist. https://github.com/serverless/serverless-kubeless/issues/25

Note that keys can be added at the provider level to make all functions in the service inherit them. Or they can be specified per function. For instance the provider level memory could be set to 512 but then overriden at the function level.

**Comment by andresmgot** [comment]
- [x] Namespace #39
- [ ] Memory size
- [ ] Description
- [ ] Environment
- [ ] Labels
- [x] Events.trigger #40
- [ ] Events.http
- [ ] Events.schedule

**Comment by sebgoa** [comment]
after our internal discussion, it is true that you can add the description and the labels directly.
Since this is regular metadata and that you post to the function object directly.

**Comment by AndreaPravato** [comment]
I see that now https://github.com/kubeless/kubeless/issues/214 (which was blocking the last checkbox: Events.schedule) is closed

**Comment by andresmgot** [comment]
Yes, we just released the version 0.2.0 that contains the `schedule` event type. We can close this issue now :)
